### DISCUSSION TITLE
project idea on Discovery Board

### DISCUSSION THREAD

**Discussion by NitinSaxenait** [japaric/discovery#409] [General]
Hi, I have completed learning on the Discovery Board. I have completed my findings on the sensors of the F3 Board. Now I want to make a project which must involve the use of the sensors of the Board. Can someone suggest me any idea on this? Thanks :)

**Comment by sirhcel** [comment]
Hello @NitinSaxenait, congratulations for working through the Discovery Book!

The thing which made me wanting to get into Embedded Rust was this [Madgwick filter demo](https://github.com/japaric/f3) from @japaric. I love this example because it does the sensor data fusion on the device an reports back its orientation to a host application where visualization of the boards actual orientation runs.

I would like to bring this demo back to life with an up-to-date code base so that others could enjoy it. For conveniently accessing the orientation data on the host, I would come in pretty handy to provide the orientation data via a USB serial device and make the visualization read it from there rather than ITM (which I did not manage to get work at 2 MBaud under macOS due to the limited set of baud rates it supports with the default `ioctl`).

I got the original code [working somehow](https://github.com/sirhcel/f3/tree/madgwick-1.44) some years ago. But I got stuck lifting this onto the current [`stm32f3-discovery`](https://github.com/rubberduck203/stm32f3-discovery) HAL while [working on the USB transfers](https://github.com/TheZoq2/hal-discovery/tree/wip-f3discovery/f3discovery/src/bin).

I have currently no time to work on this. But may be this is something appealing for you ...
